src/tools/compiletest/src/runtest.rs

```diff
@@ -1987,9 +1996,21 @@ impl<'test> TestCx<'test> {
                 rustc.arg("-Cstrip=debuginfo");
             }
             MirOpt => {
+                let app;
+                if passes.is_some() {
+                    if passes.clone().unwrap().len() == 1 {
+                        app = passes.unwrap()[0].clone();
+                    } else {
+                        app = passes.unwrap().join(" | ");
+                    }
+                } else {
+                    app = "all".to_string();
+                }
```

Comment:
```suggestion
                let app = if passes.is_some() {
                    if passes.clone().unwrap().len() == 1 {
                        passes.unwrap()[0].clone()
                    } else {
                        passes.unwrap().join(" | ")
                    }
                } else {
                    "all".to_string()
                }
```

Reply:
Also: is the `len` check needed? `join` should work on a single element vec by just grabbing the first element.

Reply:
Yes thanks did not need the len check